drivers/scheduler/dcos/dcos.go

```diff
@@ -208,24 +220,124 @@ func (d *dcos) DeleteTasks(ctx *scheduler.Context) error {
 	return nil
 }
 
-func (d *dcos) GetVolumes(ctx *scheduler.Context) ([]string, error) {
-	// TODO: Implement this method
-	return nil, nil
-}
-
 func (d *dcos) GetVolumeParameters(ctx *scheduler.Context) (map[string]map[string]string, error) {
-	// TODO: Implement this method
-	return nil, nil
+	// DC/OS does not have volume objects like Kubernetes. We get the volume information from
+	// the app spec and get the options parsed from the respective volume driver
+	volDriver, err := volume.Get(d.volDriverName)
+	if err != nil {
+		return nil, err
+	}
+
+	result := make(map[string]map[string]string)
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*marathon.Application); ok {
+			params := *obj.Container.Docker.Parameters
+			for _, p := range params {
+				if p.Key == "volume" {
+					volName, volParams, err := volDriver.ExtractVolumeInfo(p.Value)
+					if err != nil {
+						return nil, &ErrFailedToGetVolumeParams{
+							App:   ctx.App,
+							Cause: fmt.Sprintf("Failed to extract volume info: %v. Err: %v", p.Value, err),
+						}
+					}
+					result[volName] = volParams
+				}
+			}
+		}
+	}
+
+	return result, nil
 }
 
 func (d *dcos) InspectVolumes(ctx *scheduler.Context) error {
-	// TODO: Implement this method
+	// DC/OS does not have volume objects like Kubernetes. We get the volume information from
+	// the app spec and get the options parsed from the respective volume driver
+	volDriver, err := volume.Get(d.volDriverName)
+	if err != nil {
+		return err
+	}
+
+	for _, spec := range ctx.App.SpecList {
+		if obj, ok := spec.(*marathon.Application); ok {
```

Comment:
Duplicate code as above? Can this be moved into getVolumes() that can be called from here and above method?

Reply:
done